app/database/schema.py

```diff
@@ -0,0 +1,102 @@
+from sqlalchemy import Column, Integer, String, Float, Boolean
+from sqlalchemy.ext.declarative import declarative_base
+
+
+Base = declarative_base()
+
+
+class BuildStateSchema(Base):
+    """Describes the state of a Build"""
+    __tablename__ = 'builds'
+
+    build_id = Column(Integer, primary_key=True, autoincrement=True)
+    completed = Column(Boolean)
+
+
+class BuildMetaSchema(Base):
+    """Describes all the Build attributes that are basic datatypes"""
+    __tablename__ = 'build_metas'
+
+    build_id = Column(Integer, primary_key=True)
+    artifacts_tar_file = Column(String)
+    artifacts_zip_file = Column(String)
+    error_message = Column(String)
+    postbuild_tasks_are_finished = Column(String)
+    setup_failures = Column(Integer)
+    timing_file_path = Column(String)
+    # call `self.generate_project_type()` after init
+
+
+class BuildArtifactSchema(Base):
+    """Describes the _build_artifact attribute of a Build"""
+    __tablename__ = 'build_artifacts'
+
+    build_id = Column(Integer, primary_key=True)
+    build_artifact_dir = Column(String)
+
+
+class FailedArtifactDirectoriesSchema(Base):
+    """Describes the list of _failed_artifact_directories (List[str]) from BuildArtifact"""
+    __tablename__ = 'failed_artifact_directories'
+
+    uid = Column(Integer, primary_key=True, autoincrement=True)
+    build_id = Column(Integer)
+    failed_artifact_directory = Column(String)
+
+
+class FailedSubjobAtomPairsSchema(Base):
+    """Describes the list of _failed_subjob_atom_pairs (List[int, int]) from BuildArtifact"""
+    __tablename__ = 'failed_subjobs_atom_pairs'
+
+    uid = Column(Integer, primary_key=True, autoincrement=True)
```

Comment:
Yeah we've used `id` as a db field in other services and it hasn't caused any issues. Usages of it would nearly always be namespaced in an instance or class reference anyway so it wouldn't override the builtin.